Code/Framework/AzCore/Tests/DOM/DomPrefixTreeTests.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/DOM/DomPrefixTree.h>
+#include <Tests/DOM/DomFixtures.h>
+
+namespace AZ::Dom::Tests
+{
+    using DomPrefixTreeTests = DomTestFixture;
+
+    TEST_F(DomPrefixTreeTests, GetAndSetRoot)
+    {
+        DomPrefixTree<AZStd::string> tree;
+        tree.SetValue(Path(), "root");
+        EXPECT_EQ(*tree.ValueAtPath(Path(), PrefixTreeMatch::ExactPath), "root");
+    }
+
+    TEST_F(DomPrefixTreeTests, GetExactPath)
+    {
+        DomPrefixTree<int> tree;
+
+        tree.SetValue(Path("/foo/0"), 0);
+        tree.SetValue(Path("/foo/1"), 42);
+        tree.SetValue(Path("/foo/foo"), 1);
+        tree.SetValue(Path("/foo/bar"), 2);
+
+        EXPECT_EQ(*tree.ValueAtPath(Path("/foo/0"), PrefixTreeMatch::ExactPath), 0);
+        EXPECT_EQ(*tree.ValueAtPath(Path("/foo/1"), PrefixTreeMatch::ExactPath), 42);
+        EXPECT_EQ(*tree.ValueAtPath(Path("/foo/foo"), PrefixTreeMatch::ExactPath), 1);
+        EXPECT_EQ(*tree.ValueAtPath(Path("/foo/bar"), PrefixTreeMatch::ExactPath), 2);
+
+        EXPECT_EQ(tree.ValueAtPath(Path(), PrefixTreeMatch::ExactPath), nullptr);
+        EXPECT_EQ(tree.ValueAtPath(Path("/foo"), PrefixTreeMatch::ExactPath), nullptr);
+        EXPECT_EQ(tree.ValueAtPath(Path("/foo/0/subpath"), PrefixTreeMatch::ExactPath), nullptr);
```

Comment:
Normally the expected value is supplied as the first parameter.

Reply:
...so it is, there's some other Generic DOM tests I'll update as time permits.